lib/core.js

```diff
@@ -300,6 +300,15 @@ interface Iterable<T> {
     @@iterator(): Iterator<T>;
 }
 
+/* Generators */
+interface Generator<Y,R,N> {
+    @@iterator(): Iterator<Y>;
+    next(value?: N): IteratorResult<R|Y>;
```

Comment:
If I change the union here from `R|Y` to `Y|R`, the test outcomes change. Tests that currently exhibit expected failures start to succeed unexpectedly. It almost seems like the speculative match failure in the first branch isn't working—instead of falling through to the other case, we see an error right away.